targets/spidermonkey/templates/layout_head.c

```diff
@@ -28,8 +28,13 @@ static bool dummy_constructor(JSContext *cx, uint32_t argc, jsval *vp) {
         typeClass = typeMapIter->second;
         CCASSERT(typeClass, "The value is null.");
 
+\#if (COCOS2D_VERSION >= 0x00031000)
```

Comment:
Since the spider monkey version is the same, we have to use cocos2d-x version to identify the difference.